pallets/mint_with_fees/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+pub use frame_support::{
+        dispatch::DispatchResult, pallet_prelude::*,
+        traits::Currency,
+        sp_runtime::traits::{CheckedConversion, CheckedDiv}
+    };
+pub use frame_system::{
+        pallet_prelude::*,
+        ensure_root,
+    };
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        type Currency: Currency<Self::AccountId>;
+	}
+
+    // type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+    // type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// #[pallet::storage]
+	// #[pallet::getter(fn something)]
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// the value have been minted on the target account
+		/// [target_account, value]
+		ValueMinted(T::AccountId, u128),
+		/// the fees have been minted on the nsm account
+		/// [nsp_account, value]
+		FeesMinted(T::AccountId, u128),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+        /// Overflow when adding the value to the account
+		Overflow,
+        /// Amount is 0
+        ZeroAmount,
+	}
+
+    #[pallet::storage]
+    #[pallet::getter(fn percent_for_nsp)]
+    pub type PercentForNSM<T: Config> = StorageValue<_, u8>;
+
+    #[pallet::genesis_config]
+	pub struct GenesisConfig {
+		pub percent_for_nsp: u8
+	}
+
+	#[cfg(feature = "std")]
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self {
+				percent_for_nsp: 10,
+			}
+		}
+	}
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+            PercentForNSM::<T>::put(self.percent_for_nsp);
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+        ///mint value to the target account and fees to the network_service_provider account
+		#[pallet::weight(10_000)]
+		pub fn mint(
+            origin: OriginFor<T>,
+			target_account: T::AccountId,
+			network_service_provider: T::AccountId,
+            amount: u128,
+		) -> DispatchResult {
+            ensure_root(origin)?;
+            if amount == 0 {
+                return Err(Error::<T>::ZeroAmount.into());
+            }
+
+            let fee_amount: u128 = amount.checked_div(Self::percent_for_nsp().unwrap() as u128).unwrap();
```

Comment:
Take the habit to always explain why `unwrap` is safe to call in this specific place. In comments.

Using ValueQuery will remove the need for the first unwrap.
And you should handle the second one and return an error. Somebody may edit your code later on, removing the guard you put (that you forgot to put in this case) and we don't want to panic